tensorflow/core/util/gpu_device_functions.h

```diff
@@ -192,11 +192,14 @@ __device__ const unsigned kGpuWarpAll = 0xffffffff;
 __device__ inline unsigned GpuLaneId() {
   unsigned int lane_id;
 #if GOOGLE_CUDA
-#if __clang__
-  return __nvvm_read_ptx_sreg_laneid();
-#else   // __clang__
+//#if __clang__
```

Comment:
That's surprising. What clang version are you using?

Reply:
I'm using XCode 9.4.1. Here is the detail:
```
iMac18:data tomheaven$ clang++ --version
Apple LLVM version 9.1.0 (clang-902.0.39.2)
Target: x86_64-apple-darwin17.7.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

Reply:
Any suggestions?

Reply:
I re-write this part and provide a separate version for macOS. At least the change does break things.